linera-core/src/client/mod.rs

```diff
@@ -448,6 +453,42 @@ impl<Env: Environment> Client<Env> {
             .collect())
     }
 
+    async fn fetch_blob(
+        &self,
+        blob_id: BlobId,
+        nodes: &[RemoteNode<impl ValidatorNode>],
+    ) -> Result<Blob, LocalNodeError> {
+        // TODO(#3844): a very naive algorithm, just trying the validator nodes one by one - to be
+        // replaced with something better
+        for node in nodes {
+            let blob_certificate = match node.download_certificate_for_blob(blob_id).await {
+                Ok(certificate) => certificate,
+                Err(_) => continue,
+            };
+            let Some(blob) = blob_certificate
+                .value()
+                .block()
+                .created_blobs()
```

Comment:
Since this function doesn't assume it's a chain description blob, we shouldn't assume it's in `created_blobs`. Could just be a _required_ blob.